src/core.c/IO/Path.pm6

```diff
@@ -362,18 +362,15 @@ my class IO::Path is Cool does IO {
               !! self
         }
     }
-    multi method parent(IO::Path:D:) {    # XXX needs work
+    multi method parent(IO::Path:D:) {
         my $curdir := $!SPEC.curdir;
         my $updir  := $!SPEC.updir;
 
-        nqp::clone(self).cloned-with-path: self.is-absolute
-          ?? $!SPEC.join($.volume, $.dirname, '')
-          !! $.dirname eq $curdir && $.basename eq $curdir
-            ?? $!SPEC.join($.volume,$curdir,$updir)
-            !! ($.dirname eq $curdir && $.basename eq $updir)
-                 || !$!SPEC.splitdir($.dirname).first(* ne $updir)
-              ?? $!SPEC.join($.volume,$!SPEC.catdir($.dirname,$updir),$.basename)
-              !! $!SPEC.join($.volume, $.dirname, '')
+        nqp::clone(self).cloned-with-path: $.basename eq $curdir
+            ?? $!SPEC.join($.volume,$.dirname,$updir)
+            !! $.basename eq $updir
```

Comment:
Since `$.basename` is rather expensive, I would probably keep that in a `my $basename := $.basename` for performance.

Reply:
Thanks for the hint. Done!